unpackbootimg.c

```diff
@@ -191,7 +191,7 @@ int main(int argc, char **argv)
     fseek(f, i, SEEK_SET);
     if(fread(&header, sizeof(header), 1, f)) {};
 
-    if(magic == BOOT_MAGIC) {
+    if(strcmp(magic, BOOT_MAGIC)) {
```

Comment:
This results in:
```
$ ./unpackbootimg -i boot.img -o test
ANDROID! magic found at: 0
BOARD_VENDOR_CMDLINE
BOARD_VENDOR_BASE 0x17ff815a
BOARD_NAME
BOARD_PAGE_SIZE 1379272
BOARD_KERNEL_OFFSET 0x00008000
BOARD_RAMDISK_OFFSET 0xe80084d2
BOARD_TAGS_OFFSET 0xe8007ea6
BOARD_HEADER_VERSION 11747570
BOARD_HEADER_SIZE 0
BOARD_DTB_SIZE 0
BOARD_DTB_OFFSET 0xffffffffe8007ea6

$ ./unpackbootimg -i vendor_boot.img -o test
VNDRBOOT magic found at: 0
BOARD_KERNEL_CMDLINE .console=ttyMSM0 printk.devkmsg=on msm_rtb.filter=0x237 ehci-hcd.park=3 service_locator.enable=1 androidboot.memcg=1 cgroup.memory=nokmem lpm_levels.sleep_disabled=1 usbcore.autosuspend=7 androidboot.usbcontroller=a600000.dwc3 swiotlb=2048 androidboot.boot_devices=soc/1d84000.ufshc loop.max_part=7 snd_soc_cs35l41_i2c.async_probe=1 i2c_qcom_geni.async_probe=1 st21nfc.async_probe=1 spmi_pmic_arb.async_probe=1 ufs_qcom.async_probe=1 buildvariant=user
BOARD_KERNEL_BASE 0xffff9000
BOARD_NAME 00n8 androidboot.console=ttyMSM0 printk.devkmsg=on msm_rtb.filter=0x237 ehci-hcd.park=3 service_locator.enable=1 androidboot.memcg=1 cgroup.memory=nokmem lpm_levels.sleep_disabled=1 usbcore.autosuspend=7 androidboot.usbcontroller=a600000.dwc3 swiotlb=2048 androidboot.boot_devices=soc/1d84000.ufshc loop.max_part=7 snd_soc_cs35l41_i2c.async_probe=1 i2c_qcom_geni.async_probe=1 st21nfc.async_probe=1 spmi_pmic_arb.async_probe=1 ufs_qcom.async_probe=1 buildvariant=user
BOARD_PAGE_SIZE 1299805300
BOARD_HASH_TYPE sha1
BOARD_KERNEL_OFFSET 0x00008000
BOARD_RAMDISK_OFFSET 0x01007000
BOARD_SECOND_OFFSET 0x736edf63
BOARD_TAGS_OFFSET 0x3d65dc6f
BOARD_OS_VERSION 25.13.38
BOARD_OS_PATCH_LEVEL 2019-01
BOARD_DT_SIZE 741363027
Segmentation fault (core dumped)
```

So I think you meant `!strcmp(` instead, since it appears your change will reverse the intended logic, causing it to treat bootimg files as vndrboot, and vice versa.

Reply:
This is how strcmp is previously used in the code, which is also correct, stylistically you should match it for consistency:

https://github.com/osm0sis/mkbootimg/blob/master/unpackbootimg.c#L136